app/src/main/java/com/example/myweatherapp/viewHolders/NewsHolder.kt

```diff
@@ -6,47 +6,38 @@ import com.bumptech.glide.Glide
 import com.example.myweatherapp.R
 import com.example.myweatherapp.databinding.NewsHolderBinding
 import com.example.myweatherapp.interfaces.OnItemClickListener
+import com.example.myweatherapp.interfaces.OnSourceClickListener
 import com.example.myweatherapp.models.NewsModel
+import java.time.format.DateTimeFormatter
 
 
-class NewsHolder(private val newsListener: OnItemClickListener, itemView: View) : RecyclerView.ViewHolder(itemView) {
+class NewsHolder(private val newsListener: OnItemClickListener, private val sourceListener: OnSourceClickListener, itemView: View) : RecyclerView.ViewHolder(itemView) {
 
     private val binding = NewsHolderBinding.bind(itemView)
 
-    fun bind(news: NewsModel) = with(binding) {
+    fun bind(news: NewsModel, favoriteUrls: List<String>) = with(binding) {
+        val dateTimeFormatter = DateTimeFormatter.ofPattern("dd.MM.yyyy")
```

Comment:
please extract it to a global variable. now you're making a new instance again and again in the method

Reply:
dateTimeFormatter was extracted to a global variable